### PR TITLE
test(pipenv): Rewrite test mocks

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

Prepare the codebase for usage of external `detect-tools` package: mock `fs-extra` instead of internal fs utils.

This should make our extracted code to act as drop-in replacement with no other changes required. The alternative could be mocking of the `detect-tools` itself, but in this case we are losing test coverage (while pretending we have it).

## Context

- Ref: #29000

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
🤔 looks like a step back from our tooling best practices. we explicitly want to not mock fs-extra to allow partial real fa reads.

**Comment by zharinov** [comment]
1. 
```
pipenv <- fs <- fs-extra
          ^^
          Good: mock our own code which we sure works fine
```

2. 
```
pipenv <- detect-tools <- fs-extra
          ^^^^^^^^^^^^
          Bad: mock the code we never run on production
               and we not sure if it's correct
```

3.
```
pipenv <- fs <- fs-extra
                ^^^^^^^^
                This PR: mock the underlying level to make sure
                new external tool behaves similar
```

4.
```
pipenv <- detect-tool <- fs-extra
          ^^^^^^^^^^^    ^^^^^^^^
          Next PR: the implementation is replaced,
                   but tests are still working because
                   we target the underlying level
```

File: lib/modules/manager/pipenv/artifacts.spec.ts [hunk 1]
```diff
@@ -136,16 +158,31 @@ describe('modules/manager/pipenv/artifacts', () => {
...
+
+    mockFiles({
+      '/Pipfile.lock': JSON.stringify({
+        _meta: {
+          requires: { python_full_version: '3.7.6' },
+        },
+      } satisfies PipfileLockSchema),
+    });
+
+    fsExtra.ensureDir.mockResolvedValue(undefined as never);
```
[the remark this answers is not in the document]
Comment (zharinov) [reply lib/modules/manager/pipenv/artifacts.spec.ts:183]: No, it will be a type error

**Review by rarkins — DISMISSED** [review]

**Comment by rarkins** [comment]
@zharinov failures on windows:

```
    - Expected  - 3
    + Received  + 3

      Array [
        Array [
    -     "/tmp/renovate/cache/others/pipenv",
    +     "D:/tmp/renovate/cache/others/pipenv",
        ],
        Array [
    -     "/tmp/renovate/cache/others/pip",
    +     "D:/tmp/renovate/cache/others/pip",
        ],
        Array [
    -     "/tmp/renovate/cache/others/virtualenvs",
    +     "D:/tmp/renovate/cache/others/virtualenvs",
        ],
      ]

      910 |     ]);
      911 |
    > 912 |     expect(fsExtra.ensureDir.mock.calls).toEqual([
          |                                          ^
      913 |       ['/tmp/renovate/cache/others/pipenv'],
      914 |       ['/tmp/renovate/cache/others/pip'],
      915 |       ['/tmp/renovate/cache/others/virtualenvs'],
```

**Review by rarkins — DISMISSED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**